src/test/java/core/basesyntax/services/impl/CsvFileReaderServiceImplTest.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.services.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.exceptions.NoSuchFileEx;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CsvFileReaderServiceImplTest {
+    private static CsvFileReaderServiceImpl readerService;
+    private static Path validPath;
+    private static Path invalidPath;
+    private static List<String> testData;
+
+    @BeforeClass
+    public static void beforeClass() {
+        validPath = Path.of("src/test/resources/CsvFileReaderServiceImplTestValid.csv");
+        invalidPath = Path.of("CsvFileReaderServiceImplTestValid.csv");
+        readerService = new CsvFileReaderServiceImpl();
+        testData = List.of("line1, 1, 2, 3",
+                "line2, 1, 2, 3",
+                "line3, 1, 2, 3",
+                "line4, 1, 2, 3");
+    }
+
+    @Test
+    public void readFromFile_validPath_ok() throws IOException {
```

Comment:
you have 2 cases inside of single method, split them
\+ asserting can be done like this, and that would be enough
```
assertEquals(expected.size(), actual.size())
for (...) {
    assertEquals(expected.get(i), actual.get(i));
}
```

Reply:
++